tests/ui/ignored_unit_patterns.rs

```diff
@@ -13,3 +13,13 @@ fn main() {
     if let Ok(_) = foo() {}
     let _ = foo().map_err(|_| todo!());
 }
+
+#[allow(unused)]
+pub fn moo(_: () /* Do not lint */) {
+    // Lint
+    let _ = foo().unwrap();
+    // Do not lint
+    let _: () = foo().unwrap();
+    // Do not lint
+    let _: () = ();
+}
```

Comment:
Can we use `ui_test` error annotations here? Like:
```suggestion
pub fn moo(_: ()) {
    let _ = foo().unwrap();
    //~^ ERROR: blah blah blah
    let _: () = foo().unwrap();
    let _: () = ();
}
```
Can you also add them to the already existing tests as well?

Reply:
Done (in a separate and prior commit for the existing tests).